packages/custom-webpack/src/utils.ts

```diff
@@ -1,15 +1,24 @@
+import { resolve } from 'path';
+
 /**
- * check for TS node registration
- * @param file: file name or file directory are allowed
- * @todo tsNodeRegistration: require ts-node if file extension is TypeScript
+ * Register `ts-node` if webpack config file is a TypeScript file.
+ *
+ * @param file - webpack config file path
+ * @param root - path to project root
  */
-export function tsNodeRegister(file: string = '') {
-  if (file && file.endsWith('.ts')) {
-    // Register TS compiler lazily
-    require('ts-node').register({
-      compilerOptions: {
-        module: 'commonjs',
-      },
-    });
+export function tsNodeRegister(file = '', root = process.cwd()) {
+  if (!file.endsWith('.ts')) {
+    return;
   }
+
+  const tsConfigPath = resolve(root, 'tsconfig.app.json');
+
+  const { register } = require('ts-node');
+
+  register({
+    project: tsConfigPath,
+    compilerOptions: {
```

Comment:
If we provide the project then maybe we don't need the `compilerOptions` here?

Reply:
I am not quite sure if `project` overrides it but in one of my projects I have `"module": "ES2020",`.